spec/keisan/default_functions_spec.rb

```diff
@@ -93,13 +101,21 @@
           expect(Keisan::Calculator.new.evaluate("filter([-1,0,1], x, x > 0)")).to eq [1]
           expect(Keisan::Calculator.new.evaluate("select([1,2,3,4], x, x % 2 == 0)")).to eq [2,4]
           expect(Keisan::Calculator.new.simplify("[1,3,5].filter(x, x == 3)").to_s).to eq "[3]"
+          expect(Keisan::Calculator.new.evaluate("l.filter(x, x['a'] == 2)", l: [{ 'a' => 2 }, { 'a' => 3 }]))
+            .to eq([{ 'a' => 2 }])
+          expect(Keisan::Calculator.new.evaluate("h['l'].filter(x, x['a'] == 2)", h: { 'l' => [{ 'a' => 2 }, { 'a' => 3 }] }))
+            .to eq([{ 'a' => 2 }])
```

Comment:
I see that here we are checking that we can run `filter` on an element of a hash. Could you also add a test that runs `filter` on an element of a list too?